src/lib.rs

```diff
@@ -0,0 +1,219 @@
+extern crate regex;
+
+use regex::Regex;
+use std::collections::HashMap;
+
+#[derive(Debug, PartialEq)]
+pub enum IdentType {
+    Space,
+    Tab,
+    None
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Indent {
+    amount: usize,
+    indent: String,
+    ident_type: IdentType
+}
+
+fn ensure_indent_isize(x: usize) -> isize {
+    if x > std::isize::MAX as usize {
+        panic!("Indent greater then std::isize::MAX");
+    } else {
+        x as isize
+    }
+}
+
+fn get_most_used(indents: &HashMap<isize, (isize, isize)>) -> usize {
```

Comment:
I think you'd make the code a little more readable and clear by changing this tuple to a struct type so you can have nice names for the two elements. That will especially make the `.0` and `.1` parts below more readable. Of course you're porting this code so maybe it's not obvious to me what they mean :P